File: 20140206/fizzbuzz.rb
class FizzBuzz
  def initialize min, max
    @min = min
    @max = max
  end

  def fizz? i
    i % 3 == 0
  end
  private :fizz?

  def buzz? i
    i % 5 == 0
  end
  private :buzz?

  def fizzbuzz? i
    s = ""
    s += "Fizz" if fizz? i
    s += "Buzz" if buzz? i
    s += i.to_s if s==""
    return s
  end
  private :fizzbuzz?

  def run
    for i in @min..@max
      puts fizzbuzz? i
    end
  end
end

FizzBuzz.new(1, 100).run

File: 20140206/fizzbuzz.hs
module Main (main) where

isFizz :: Int -> Bool
isFizz i =
    i `mod` 3 == 0

isBuzz :: Int -> Bool
isBuzz i =
    i `mod` 5 == 0

toStr :: Int -> String
toStr i =
      if (isFizz i) then
         if (isBuzz i) then
            "FizzBuzz"
         else
            "Fizz"
      else
        if (isBuzz i) then
           "Buzz"
        else
           show i

fizzbuzz :: Int -> [String]
fizzbuzz n = map toStr [1 .. n]

main :: IO ()
main = do
     print (fizzbuzz 100)